x-pack/plugins/cloud_security_posture/public/common/api/use_license_management_locator_api.tsx

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+import { useKibana } from '../hooks/use_kibana';
+
+const LICENSE_MANAGEMENT_LOCATOR = 'LICENSE_MANAGEMENT_LOCATOR';
+
+const getLicenseManagementLocatorKey = 'license_management_url_key';
+
+export const useLicenseManagementLocatorApi = () => {
+  const { share } = useKibana().services;
+
+  return useQuery([getLicenseManagementLocatorKey], () => {
+    const locator = share.url.locators.get(LICENSE_MANAGEMENT_LOCATOR);
+    if (!locator) return;
+
+    return locator.getUrl({
+      page: 'dashboard',
+    });
+  });
```

Comment:
worth adding a comment explaining why is it optional, emphasizing this URL does not exist on serverless